File: android-library/src/main/java/com/zetapush/library/ZetaPushConnectionEvent.java
package com.zetapush.library;

import java.util.Map;

/**
 * Created by mikaelmorvan on 06/02/2018.
 */

/**
 * Callback interface for ZetaPush connection.
 */
public interface ZetaPushConnectionEvent {

    void successfulHandshake(Map<String, Object> map);

    void failedHandshake(Map<String, Object> map);

    void connectionEstablished();

    void connectionBroken();

    void connectionClosed();

    void messageLost(String s, Object o);
}

File: android-library/src/main/java/com/zetapush/library/ZetaPushConnectionEvent.kt
package com.zetapush.library

/**
 * Callback interface for ZetaPush connection.
 */
interface ZetaPushConnectionEvent {

    fun successfulHandshake(map: Map<String, Any>?)

    fun failedHandshake(map: Map<String, Any>?)

    fun connectionEstablished()

    fun connectionBroken()

    fun connectionClosed()

    fun messageLost(s: String?, o: Any?)
}
